diff --git a/meson.build b/meson.build
index 633d821..23b0244 100644
--- a/meson.build
+++ b/meson.build
@@ -44,7 +44,6 @@ gtk = dependency('gtk+-3.0', version: '>=3.20.0')
 gio = dependency('gio-2.0')
 
 subdir('res')
-subdir('src/po')
 
 executable(
 	'swappy',
diff --git a/src/application.c b/src/application.c
index fcbdbe5..f46e9b7 100644
--- a/src/application.c
+++ b/src/application.c
@@ -686,9 +686,6 @@ static bool load_layout(struct swappy_state *state) {
   /* Construct a GtkBuilder instance and load our UI description */
   GtkBuilder *builder = gtk_builder_new();
 
-  // Set translation domain for the application based on `src/po/meson.build`
-  gtk_builder_set_translation_domain(builder, GETTEXT_PACKAGE);
-
   if (gtk_builder_add_from_resource(builder, "/me/jtheoof/swappy/swappy.glade",
                                     &error) == 0) {
     g_printerr("Error loading file: %s", error->message);
