diff --git a/ziguard.awk b/ziguard.awk
index 0728baa..9b3f2f8 100644
--- a/ziguard.awk
+++ b/ziguard.awk
@@ -165,7 +165,7 @@ DATAFORM != "main" {
     sub(/[-+][^\t ]+CHANGE-TO-/, "")
   } else {
     if (/^[^#]*%z/) {
-      stdoff_column = 2 * /^Zone/ + 1
+      stdoff_column = 2 * int(/^Zone/) + 1
       rules_column = stdoff_column + 1
       stdoff = get_minutes($stdoff_column)
       rules = $rules_column
@@ -222,7 +222,7 @@ DATAFORM != "main" {
       stdoff_subst[1] = rounded_stdoff
     }
   } else if (stdoff_subst[0]) {
-    stdoff_column = 2 * /^Zone/ + 1
+    stdoff_column = 2 * int(/^Zone/) + 1
     stdoff_column_val = $stdoff_column
     if (stdoff_column_val == stdoff_subst[0]) {
       sub(stdoff_subst[0], stdoff_subst[1])
